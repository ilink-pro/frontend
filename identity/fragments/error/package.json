{
  "name": "@identity/error-fragment",
  "version": "0.0.1",
  "license": "BSD-3-Clause",
  "main": "src/index.tsx",
  "devDependencies": {
    "@atls/next-identity-integration": "^0.0.8",
    "@emotion/react": "^11.9.0",
    "@emotion/styled": "^11.8.1",
    "@ilink-ui-proto/button": "0.0.1",
    "@types/react": "^17.0.38",
    "@types/styled-system": "^5.1.15",
    "@ui/layout": "workspace:0.0.1",
    "@ui/text": "workspace:0.0.1",
    "styled-system": "^5.1.5",
    "styled-tools": "^1.7.2"
  },
  "peerDependencies": {
    "react": "^17.0.1",
    "react-intl": "^5.18.1"
  }
}
