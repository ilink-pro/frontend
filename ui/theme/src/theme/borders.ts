const borders = {
  soft: '1px solid #E3DEF0',
}

export { borders }
