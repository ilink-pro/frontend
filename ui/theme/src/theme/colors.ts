const colors = {
  background: {
    white: 'rgb(255,255,255)',
    lightPurple: 'rgb(251,250,255)',
    lightPurpleAccent: 'rgb(219,216,238)',
    purpleAccent: 'rgb(116,97,240)',
  },
  text: {
    primary: 'rgb(54,52,61)',
    accent: 'rgb(116,97,240)',
    secondary: 'rgb(153,147,189)',
    red: 'rgb(220,103,103)',
    gray: 'rgb(155,164,167)',
    white: 'rgb(255,255,255)',
  },
  button: {
    secondary: {
      default: {
        background: 'rgb(255,255,255)',
        border: 'rgb(219,216,238)',
        font: 'rgb(116,97,240)',
      },
      hover: {
        background: 'rgb(255,255,255)',
        border: 'rgb(116,97,240)',
        font: 'rgb(116,97,240)',
      },
      pressed: {
        background: 'rgb(255,255,255)',
        border: 'rgb(231,228,251)',
        font: 'rgb(116,97,240)',
      },
      disabled: {
        background: 'rgb(255,255,255)',
        border: 'rgb(219,216,238)',
        font: 'rgb(153,147,189)',
      },
    },
  },
  select: {
    menu: {
      border: 'rgb(219,216,238)',
    },
    item: {
      default: {
        background: 'rgb(255,255,255)',
      },
      hover: {
        background: 'rgb(247,246,254)',
      },
    },
  },
}

export { colors }
