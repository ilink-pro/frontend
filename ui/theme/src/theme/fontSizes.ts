export const fontSizes = {
  small: 12,
  semiRegular: 14,
  default: 16,
  increased: 18,
  semiGiant: 24,
  semiBig: 28,
  huge: 32,
  big: 36,
  giant: 40,
  colossal: 64,
}
