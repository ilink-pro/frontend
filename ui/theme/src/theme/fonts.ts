const fonts = {
  primary: 'Khula',
}

export { fonts }
