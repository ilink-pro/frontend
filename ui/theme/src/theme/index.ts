export * from './fontWeights'
export * from './lineHeights'
export * from './fontSizes'
export * from './colors'
export * from './fonts'
export * from './radii'
export * from './shadows'
export * from './borders'
