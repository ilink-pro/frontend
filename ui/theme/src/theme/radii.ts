const radii = {
  normal: 8,
  semiMedium: 12,
  medium: 16,
  full: '100%',
}

export { radii }
