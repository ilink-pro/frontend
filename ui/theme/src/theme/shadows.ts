const shadows = {
  bottom: '0px 4px 24px rgba(0, 0, 0, 0.05)',
  light: '0px 7px 16px rgba(136, 165, 191, 0.12)',
}

export { shadows }
