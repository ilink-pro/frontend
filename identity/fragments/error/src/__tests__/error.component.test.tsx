/**
 * @jest-environment jsdom
 */

import { render }        from '@testing-library/react'

import React             from 'react'
import { IntlProvider }  from 'react-intl'

import { ThemeProvider } from '@ui/theme'

import { Error }         from '../error.component'

describe('identity', () => {
  describe('error', () => {
    describe('error.component', () => {
      it('should match the latest render snapshot', () => {
        const { asFragment } = render(
          <IntlProvider locale='ru' messages={{}}>
            <ThemeProvider>
              <Error />
            </ThemeProvider>
          </IntlProvider>
        )

        expect(asFragment()).toMatchSnapshot()
      })
    })
  })
})
