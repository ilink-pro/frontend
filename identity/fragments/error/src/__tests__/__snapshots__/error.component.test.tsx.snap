// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`identity error error.component should match the latest render snapshot 1`] = `
<DocumentFragment>
  .emotion-0 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-1 {
  box-sizing: border-box;
  max-width: 448px;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-2 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-basis: 70px;
  -ms-flex-preferred-size: 70px;
  flex-basis: 70px;
}

.emotion-3 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-4 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  word-break: break-word;
  cursor: pointer;
  outline: none;
  padding: 0;
  border: 1px solid transparent;
  height: 48px;
  font-size: 14px;
  font-weight: 600;
  font-family: Khula;
  padding-left: 24px;
  padding-right: 24px;
  border-radius: 6px;
  background-color: black;
  border-color: black;
  color: white;
}

<div
    class="emotion-0"
    display="flex"
    height="100%"
  >
    <div
      class="emotion-1"
      display="flex"
      width="100%"
    />
    <div
      class="emotion-2"
      display="flex"
    />
    <div
      class="emotion-3"
      display="flex"
    >
      <a
        class="emotion-4"
        href="/auth/login"
        style="text-decoration: none;"
      >
        Go to login page
      </a>
    </div>
  </div>
</DocumentFragment>
`;
